unity/Assets/Maroon/reusableScripts/Config/ConfigLoader.cs

```diff
@@ -0,0 +1,132 @@
+        
+        public static ConfigLoader Instance => _instance;
+
+        private void Awake()
+        {
+            if (_instance == null)
+            {
+                _instance = this;
+            }
+            else if (_instance != this)
+            {
+                DestroyImmediate(this.gameObject);
+                return;
+            }
+
+            DontDestroyOnLoad(this.gameObject);
+        }
+
+        void Start()
+        {
+#if UNITY_WEBGL && !UNITY_EDITOR
+            StartCoroutine(LoadAllConfigs());
+#else
+            string basePath = Application.streamingAssetsPath + "/Config/3DMotionSimulation";
+            string[] txtFiles = Directory.GetFiles(basePath, "*.json");
+
+            _configs = new Dictionary<string, string>();
+            foreach (string file in txtFiles)
+            {
+                string json = File.ReadAllText(file);
+                string cutFilename = file.Replace(basePath + "\\", "");
+                cutFilename = cutFilename.Replace(".json", "");
+                
+                _configs.Add(cutFilename, json);
+            }
+
+            ChangeConfig("Default");
+#endif
+        } 
+
+
+#if UNITY_WEBGL && !UNITY_EDITOR
+        IEnumerator LoadAllConfigs()
+        {
+            string basePath = "http://localhost:8000/StreamingAssets/Config/3DMotionSimulation/";
+            List<string> httpFiles = new List<string>();
+            UnityWebRequest uwr = UnityWebRequest.Get("http://localhost:8000/configs.php");
+            yield return uwr.SendWebRequest();
+
+            var jsonFile = uwr.downloadHandler.text;
+            var parseJSON = JSON.Parse(jsonFile);
+
+            for(int i = 0; i < parseJSON.Count; i++) {
+                httpFiles.Add(basePath + parseJSON[i]);
+            }
+
+            for(int i = 0; i < httpFiles.Count; i++) {
+                UnityWebRequest webReq = UnityWebRequest.Get(httpFiles[i]);
+                yield return webReq.SendWebRequest();
+
+                var json = webReq.downloadHandler.text;
+                string cutFilename = httpFiles[i].Replace(basePath, "");
+                cutFilename = cutFilename.Replace(".json", "");
+
+                _configs.Add(cutFilename, json);
+            }
+            
+            if(Maroon.GlobalEntities.BootstrappingManager.Instance.UrlParameters.TryGetValue(WebGlUrlParameter.Config, out string config))
+            {
+                if (!ChangeConfig(config))
+                {
+                    ChangeConfig("Default");
+                }
+            }
+            else
+            {
+                ChangeConfig("Default");
+            }
+        }
+#endif
+
+        /// <summary>
+        /// Changes current config to the one with the given name.
+        /// Returns true if the config was found and changed, false otherwise.
+        /// </summary>
+        public bool ChangeConfig(string configName)
+        {
+            foreach (var key in _configs.Keys)
```

Comment:
If I do it like that, I think there is no way of knowing what the actual key looks like.
E.g. if configName = "balLiNtHewiNd" and the key would be "BallInTheWind", it would not say key exists, since "ballinthewind" != "BallInTheWind".

Reply:
ah hmm, didn't think about that, you're right it's probably the easiest to maintain variant that works and is easy to read as it is now